js/ui/messageTray.js

```diff
@@ -552,6 +552,8 @@ class NotificationBanner extends Calendar.NotificationMessage {
 
         this._buttonBox.add(button);
         button.connect('clicked', () => {
+            callback();
```

Comment:
Can't we just save the value of this.notification.resident here before invoking callback and use if after invoking the callback to decide whether to emit "done-displaying"?

Something like (pseudo code):
```
let resident = this.notification && this.notification.resident;
callback();
if (!resident) {
  this.emit('done-displaying');
  if (this.notification)
    this.notification.destroy();
}
```

Reply:
Ok, talked to @uajain about it and this won't work because NotificationBanner (`this` itself) gets deleted when callback is invoked, so lets merge it as is given no one is currently handling the `done-displaying` signal and follow-up with upstream on the issue.

Note that we will still get the backtrace in the journal but things should still work as expected.